Issue/35 Pull request

agylling opened the pull request:
Feature, adds the request_push function

Fixes #35

agylling committed:
feat #36 Add request_push function

Added the core to request_push function and a test case that should be succesful, having troubles with kth SMTP server. Will enter more test cases when I know what to do in order to continue

agylling committed:
feat #35 Add test cases

Added two more test cases where it fails on building for one and testing on the other

Fixes #35

agylling committed:
feat #35 Correct spelling

Changed name of buildSucess to buildSuccess

philippaornell reviewed (approved):
Looks good. Maybe you can create an issue that we need more test cases for this method before we merge, so we dont forget it?

agylling commented:
Don't know how many more test cases are needed though. There's one branch that is successful (the master), one where the build fails (nobuild) and one where the tests fail (testsfail) so I think all of them are covered?

mstniy commented on src/main/java/CI/GitHandler.java:
```diff
@@ -36,14 +36,38 @@
-        // TODO
-      // Called by request_push
       // Try and pull the branch
+      String path = pull_branch();
```
pull_branch may fail here (the branch may not exist or a temporary network failure may happen)
One may be inclined to think that the branch should exist at this point because someone pushed to it, but keep in mind the possible race condition that the branch was pushed to and deleted before this callback had the opportunity to run.

mstniy commented on src/test/java/CI/GitHandlerTest.java:
```diff
@@ -7,6 +7,30 @@
+    gh.request_push();
+    // Fail at test after build
+    event.setBranchName("testsfail");
+    gh.request_push();
```
We need to have assert's here. Maybe we can modify request_push such that it returns a boolean (or even more descriptive, an enum of somthing like PULL_FAILED, BUILD_FAILED, TEST_FAILED, NOTIFICATION_FAILED and so on)

mstniy commented on src/test/java/CI/GitHandlerTest.java:
```diff
@@ -7,6 +7,30 @@
+    // Fail at test after build
+    event.setBranchName("testsfail");
+    gh.request_push();
+  }
```
We also need to check for a branch that doesn't exist. The current code should crash, I believe (because pull_branch will return a null).

agylling committed:
feat #35 Improved function

After comments on pull request I've made some changes improving the code

Fixes #35

agylling committed:
Merge branch 'master' of https://github.com/top-tier-ci-coders/ContinuousIntegrationServer into Issue/35

agylling committed:
feat #35 removed redundant condition

Fixes #35

agylling committed:
feat #35 Fixed potential Bug

Fixed a potential bug in Pull_branch that would affect the request_push function

Fixes #35

mstniy reviewed (approved):
Seems good
